packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,554 @@
+    .join(' ');
+
+  return (
+    <>
+      <InputWrapper {...wrapperProps}>
+        <div className={styles.fileInputContainer}>
+          {hasDragAndDrop && (
+            <>
+              <div
+                aria-hidden
+                onDragEnter={onDragEnter}
+                onDragOver={onDragEnter}
+                onDragLeave={onDragLeave}
+                onDrop={(event: React.DragEvent<HTMLDivElement>) => {
+                  const { dataTransfer }: { dataTransfer: DataTransfer } = event;
+                  onDragLeave(event);
+                  onFilesChange(Array.from(dataTransfer.files));
+                }}
+                className={classNames(styles.dragAndDrop, isDragOverDrop && styles.dragAndDropActive)}
+                ref={dropAreaRef}
+              >
+                <div className={styles.dragAndDropLabel}>
+                  <IconUpload aria-hidden />
+                  <span className={styles.dragAndDropLabelText}>{dragAndDrop.label}</span>
+                </div>
+              </div>
+              <div className={styles.dragAndDropHelperText}>{dragAndDrop.helperText}</div>
+            </>
+          )}
+          <div className={styles.fileInputWrapper}>
+            <Button
+              aria-hidden
+              variant="secondary"
+              iconLeft={<IconPlus aria-hidden />}
+              onClick={() => passButtonClickToInput()}
+              disabled={disabled}
+            >
+              {buttonLabel}
+            </Button>
+            <input
+              type="file"
+              ref={inputRef}
+              id={id}
+              disabled={disabled}
+              required={required}
+              aria-describedby={ariaDescribedBy}
+              className={styles.fileInput}
+              onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                onFilesChange(Array.from(event.target.files));
+              }}
+              {...(accept ? { accept } : {})}
+              {...(multiple ? { multiple } : {})}
+            />
+          </div>
+        </div>
+      </InputWrapper>
+      <ul
+        id={fileListId}
+        ref={fileListRef}
+        tabIndex={-1}
+        className={styles.fileList}
+        aria-label={
+          hasFilesSelected ? getFileListAriaLabel(language, selectedFiles.length) : getNoFilesAddedMessage(language)
+        }
+      >
+        {selectedFiles.map((file: File, index: number) => (
+          <li
+            key={file.name}
+            className={styles.fileListItem}
+            tabIndex={-1}
+            ref={(el) => {
+              if (el && fileListRef.current && fileListFocusIndexRef.current === index) {
+                el.focus();
+              }
+            }}
+          >
+            {file.type.startsWith('image') ? <IconPhoto aria-hidden /> : <IconDocument aria-hidden />}
+            <div className={styles.fileListItemTitle}>
+              <span className={styles.fileListItemName}>{file.name}</span>
```

Comment:
Noticed the font-size here had a small difference from design: 16px -> 14px.

Reply:
- [ ] check correct font size